README.md

```diff
@@ -1,4 +1,201 @@
+```
+
+In the container, a2z is installed as a python "binary" on the path:
+
+```bash
+# which a2z
+/usr/local/bin/a2z
+```
+
+The all contributors client is also on the path:
+
+```bash
+$ which cli.js
+/code/node_modules/all-contributors-cli/dist/cli.js
+```
+
+### 2. Generate .all-contributorsrc
+
+Let's now change directory to where we bound our repository as a volume.
+
+```bash
+cd /data
+```
+
+If you don't have an `.all-contributorsrc` in your present working directory (the
+root of the repository) you can use [a2z.py](a2z.py) to generate an empty one. You
+can either provide the repository name directly:
+
+```bash
+$ a2z init singularityhub/sregistry
+INFO:a2z:Generating .all-contributorsrc
+```
+
+or export it to the environment as `GITHUB_REPOSITORY` (this is how it would be discovered
+during a GitHub workflow run).
+
+```bash
+export GITHUB_REPOSITORY=singularityhub/sregistry
+$ a2z init
+INFO:a2z:Generating .all-contributorsrc
+```
+In both cases, you'll generate an empty file:
+
+```bash
+$ cat .all-contributorsrc 
+{
+    "projectName": "sregistry",
+    "projectOwner": "singularityhub",
+    "repoType": "github",
+    "repoHost": "https://github.com",
+    "files": [
+        "README.md"
+    ],
+    "imageSize": 100,
+    "commit": true,
+    "commitConvention": "none",
+    "contributors": [],
+    "contributorsPerLine": 7
+}
+```
+
+### 2. Generate .zenodo.json
+
+If you want to generate a fresh Zenodo.json, you can do that as follows:
+
+```bash
+$ a2z init --zenodo 10.5281/zenodo.1012531 singularityhub/sregistry
+```
+
+While the GitHub repository is sometimes found as a related identifier,
+this isn't always the case, so you should again provide it on the command
+line or exported to `GITHUB_REPOSITORY`. By default, we parse contributors
+from the GitHub API, and include the creators already defined in Zenodo.
+You will need this `.zenodo.json` file to exist in order to update it.
+
+### 3. Update contributors
+
+We next want to use the GitHub api to discover contributors to the repository,
+for either both the `.all-contributorsrc` and the `.zenodo.json`, or just one of the two.
+
+#### Update all
+
+If you want to update both the allcontributors file and the zenodo.json, you
+can kill two birds with one stone (and essentially cache the GitHub API request)
+and do the following:
+
+```bash
+$ a2z update all
+```
+
+Note that for the update, we query the GitHub API to update both the zenodo.json
+and contributors file.
+
+#### Update allcontributors
+
+Since we already have the contributors file we don't need to provide the repository name again, however it's suggested that you export a `GITHUB_TOKEN` to increase your API limits (if necessary).
+
+```bash
+$ a2z update allcontrib
+```
+
+You can optionally set a minimum number of contributors threshold to add (defaults to 1), or
+a string to represent the kind of contribution (defaults to "core")
+
+```bash
+$ a2z update --thresh 10 --ctype docs
```

Comment:
So, would it add everyone it finds committed anything (but more than 10 commits) anywhere to `docs`?  I do not quite grasp the use case here.

Reply:
If you look at a repository like Spack, there are hundreds of contributors. They might want to set a threshold of contributions to get some top percentage. It's not perfect, because like you said commits don't always equate to quality, but for a repo that does squash and merge or similar, it would actually be reasonable.

Reply:
AAAH -- that is what it means! But then in https://github.com/con/allcontributors-zenodo/pull/2#discussion_r447136304 it should be "maximum" not minimum?... but I would not like projects which would start using this feature and slowly displace older contributors with new ones ;-)  but I guess it might make sense in some opinionated projects cases